File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.js
function solution(cards1, cards2, goal) {
    let answer = "Yes";
    
    for (const word of goal) {
        if (cards1[0] === word || cards2[0] === word) {
            if (cards1[0] === word) {
                cards1.shift();
            } else {
                cards2.shift();
            }
        } else {
            answer = "No"
            return answer
        }
    }
    
    return answer;
}

File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.py
from collections import deque

def solution(cards1, cards2, goal):
    answer = 'No'
    cards1 = deque(cards1) 
    cards2 = deque(cards2) 
    goal = deque(goal) 
    
    
    while goal:
        if cards1 and goal[0] == cards1[0]:
            cards1.popleft();
        elif cards2 and goal[0] == cards2[0]:
            cards2.popleft();
        else:
            break;
        goal.popleft()
            

    if len(goal) == 0:
        answer = 'Yes'
    
    return answer
            
        